### ISSUE TITLE
Electron: --production option is deprecated on 'install'; use 'yarn workspaces focus' instead

### ISSUE THREAD

**Issue by maggie44** [quasarframework/quasar#14618]
### What happened?

When running a build for electron it passes the --production flag which is deprecated for Yarn v2 and v3:

`yarn workspace ui build -m electron`

```
 App • Running "yarn install --production" in /Users/user/Documents/GitHub.nosync/balena-device-ui/ui/dist/electron/UnPackaged

➤ YN0050: The --production option is deprecated on 'install'; use 'yarn workspaces focus' instead

 App • Command "yarn" failed with exit code: 1

 App • ⚠️   FAIL  yarn failed installing dependencies
```

### What did you expect to happen?

Build to continue, it seems to all be working and compatible other than the flag causing an error. 

### Reproduction URL

N/A

### How to reproduce?

yarn workspace ui build -m electron

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Electron Mode

### Quasar info output

```Shell
Operating System - Darwin(21.6.0) - darwin/arm64
NodeJs - 18.9.0

Global packages
  NPM - 8.16.0
  yarn - 3.2.3
  @quasar/cli - 1.3.2
  @quasar/icongenie - 2.5.3
  cordova - Not installed

Important local packages
  quasar - 2.8.4 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.1.2 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.15.4 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.40 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.5
  pinia - 2.0.22 -- Intuitive, type safe and flexible Store for Vue
  vuex - Not installed
  vite - 2.9.15 -- Native-ESM powered web dev build tool
  eslint - 8.24.0 -- An AST-based pattern checker for JavaScript.
  electron - 21.0.1 -- Build cross platform desktop apps with JavaScript, HTML, and CSS
  electron-packager - 16.0.0 -- Customize and package your Electron app with OS-specific bundles (.app, .exe, etc.) via JS or CLI
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed
```
```


### Relevant log output

_No response_

### Additional context

_No response_

**Comment by customautosys** [comment]
> ### What happened?
> 
> When running a build for electron it passes the --production flag which is deprecated for Yarn v2 and v3:
> 
> `yarn workspace ui build -m electron`
> 
> ```
>  App • Running "yarn install --production" in /Users/user/Documents/GitHub.nosync/balena-device-ui/ui/dist/electron/UnPackaged
> 
> ➤ YN0050: The --production option is deprecated on 'install'; use 'yarn workspaces focus' instead
> 
>  App • Command "yarn" failed with exit code: 1
> 
>  App • ⚠️   FAIL  yarn failed installing dependencies
> ```
> 
> ### What did you expect to happen?
> 
> Build to continue, it seems to all be working and compatible other than the flag causing an error. 
> 
> ### Reproduction URL
> 
> N/A
> 
> ### How to reproduce?
> 
> yarn workspace ui build -m electron
> 
> ### Flavour
> 
> Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)
> 
> ### Areas
> 
> Electron Mode
> 
> ### Platforms/Browsers
> 
> _No response_
> 
> ### Quasar info output
> 
> ```Shell
> Operating System - Darwin(21.6.0) - darwin/arm64
> NodeJs - 18.9.0
> 
> Global packages
>   NPM - 8.16.0
>   yarn - 3.2.3
>   @quasar/cli - 1.3.2
>   @quasar/icongenie - 2.5.3
>   cordova - Not installed
> 
> Important local packages
>   quasar - 2.8.4 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
>   @quasar/app-vite - 1.1.2 -- Quasar Framework App CLI with Vite
>   @quasar/extras - 1.15.4 -- Quasar Framework fonts, icons and animations
>   eslint-plugin-quasar - Not installed
>   vue - 3.2.40 -- The progressive JavaScript framework for building modern web UI.
>   vue-router - 4.1.5
>   pinia - 2.0.22 -- Intuitive, type safe and flexible Store for Vue
>   vuex - Not installed
>   vite - 2.9.15 -- Native-ESM powered web dev build tool
>   eslint - 8.24.0 -- An AST-based pattern checker for JavaScript.
>   electron - 21.0.1 -- Build cross platform desktop apps with JavaScript, HTML, and CSS
>   electron-packager - 16.0.0 -- Customize and package your Electron app with OS-specific bundles (.app, .exe, etc.) via JS or CLI
>   electron-builder - Not installed
>   register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
>   @capacitor/core - Not installed
>   @capacitor/cli - Not installed
>   @capacitor/android - Not installed
>   @capacitor/ios - Not installed
> ```
> ```
> 
> 
> ### Relevant log output
> 
> _No response_
> 
> ### Additional context
> 
> _No response_

Seems like you probably need to create a GitHub repo and edit your original otherwise the automated system will close your issue. Maybe you can just create a blank quasar project, switch it to yarn 2 and put it in GitHub?

**Comment by topvis** [comment]
I'm having similar issue but it's from NPM.

After running the command ```quasar build -m electron```, I got this warning message:

```console
App • Running "npm install --production" in C:\path\app-name\dist\electron\UnPackaged

npm WARN config production Use `--omit=dev` instead.
```

It did build the package successfully. It's just a warning.


Here are my package versions:
npm: 9.2.0
electron: 22.0.0
electron-builder: 23.6.0
quasar: 2.10.2
@quasar/app-vite: 1.1.3

**Comment by devpixde** [comment]
--production option is deprecated on 'install'

I have the very same problem here. Trying to make a production build for electron app with yarn 3.3.

Is yarn 3 not supported? Maybe should give a hint on installation process for quasar not to use yarn 3.3 (if its not supported)

My solution to the problem as a workaround:
deleted yarn.lock and run npm install. Afterwards the build worked (with npm instead yarn...)

Nevertheless i would prefer to work with yarn.

**Comment by rstoenescu** [comment]
Fix will be available in q/app-vite 1.3.0 and q/app-webpack 3.8.0
